File: dotnet/console/Console/Scenarios/CreateDigitalDegreeScenario.cs
using Lacuna.Signer.Api;
using Lacuna.Signer.Api.Documents;
using Lacuna.Signer.Api.FlowActions;
using Lacuna.Signer.Api.Users;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    public class CreateDigitalDegreeScenario : Scenario
    {
        /**
         * This scenario demonstrates the creation of a digital degree compliant with "PORTARIA Nº 554, DE 11 DE MARÇO DE 2019".
         */
        public override async Task RunAsync()
        {
            // 1. The file's bytes must be read by the application and uploaded
            var filePath = "sample-degree.xml";
            var fileName = Path.GetFileName(filePath);
            var file = File.ReadAllBytes(filePath);
            var uploadModel = await SignerClient.UploadFileAsync(fileName, file, "application/xml");

            // 2. Define the name of the document which will be visible in the application
            var fileUploadModel = new FileUploadModel(uploadModel) { DisplayName = "Digital Degree Sample" };

            // 3. For each participant on the flow, create one instance of ParticipantUserModel
            var participantUserOne = new ParticipantUserModel()
            {
                Name = "Jack Bauer",
                Email = "jack.bauer@mailinator.com",
                Identifier = "75502846369"
            };

            var participantUserTwo = new ParticipantUserModel()
            {
                Name = "James Bond",
                Email = "james.bond@mailinator.com",
                Identifier = "95588148061"
            };

            var ParticipantUserThree = new ParticipantUserModel()
            {
                Name = "Gary Eggsy",
                Email = "gary.eggsy@mailinator.com",
                Identifier = "87657257008"
            };

            // 4. Specify the element that holds the namespace of the issuer
            var xmlNamespacesModel = new XmlNamespaceModel()
            {
                Prefix = "dip",
                Uri = @"http://portal.mec.gov.br/diplomadigital/arquivos-em-xsd"
            };

            // 5. The fields 'DadosDiploma' and 'DadosRegistro' and the entire XML file must be signed
            var xadesOptionsDegreeData = new XadesOptionsModel()
            {
                SignatureType = XadesSignatureTypes.XmlElement,
                ElementToSignIdentifierType = XadesElementIdentifierTypes.XPath,
                ElementToSignIdentifier = @"//dip:DadosDiploma",
                InsertionOption = XadesInsertionOptions.AppendChild
            };

            var xadesOptionsModelRegisterData = new XadesOptionsModel()
            {
                SignatureType = XadesSignatureTypes.XmlElement,
                ElementToSignIdentifierType = XadesElementIdentifierTypes.XPath,
                ElementToSignIdentifier = @"//dip:DadosRegistro",
                InsertionOption = XadesInsertionOptions.AppendChild
            };

            var xadesOptionsModelFull = new XadesOptionsModel()
            {
                SignatureType = XadesSignatureTypes.FullXml
            };

            // 6. Each signature requires its own flow action
            var degreeDataAction = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = participantUserOne,
                XadesOptions = xadesOptionsDegreeData
            };

            var registerDataAction = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = participantUserTwo,
                XadesOptions = xadesOptionsModelRegisterData
            };

            var flowActionCreateModelFull = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = ParticipantUserThree,
                XadesOptions = xadesOptionsModelFull
            };

            // 7. Send the document create request
            var documentRequest = new CreateDocumentRequest()
            {
                Files = new List<FileUploadModel>() { fileUploadModel },
                XmlNamespaces = new List<XmlNamespaceModel>() { xmlNamespacesModel },
                FlowActions = new List<FlowActionCreateModel>() 
                {
                    degreeDataAction,
                    registerDataAction,
                    flowActionCreateModelFull 
                },
            };
            var result = (await SignerClient.CreateDocumentAsync(documentRequest)).First();

            System.Console.WriteLine($"Document {result.DocumentId} created");
        }
    }
}

File: nodejs/scenarios/createDigitalDegreeScenario.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const signer_node_client_1 = require("signer-node-client");
const scenario_1 = require("./scenario");
const filePath = "../samples/sample-degree.xml";
const filename = "sample-degree.xml";
const file = (0, signer_node_client_1.getBase64)(filePath);
const uploadApi = new signer_node_client_1.UploadApi(scenario_1.config);
const documentApi = new signer_node_client_1.DocumentsApi(scenario_1.config);
// 1. The file's bytes must be read by the application and uploaded
uploadApi.apiUploadsBytesPost({
    bytes: file,
})
    .then((res) => {
    // 2. Define the name of the document which will be visible in the application
    const fileUploadModel = {
        contentType: "application/xml",
        displayName: "Digital Degree Sample",
        id: res.data.id,
        name: filename,
    };
    // 3. For each participant on the flow, create one instance of ParticipantUserModel
    const participantUserOne = {
        name: "Jack Bauer",
        email: "jack.bauer@mailinator.com",
        identifier: "75502846369",
    };
    const participantUserTwo = {
        name: "James Bond",
        email: "james.bond@mailinator.com",
        identifier: "95588148061",
    };
    const participantUserThree = {
        name: "Garry Eggsy",
        email: "garry.eggsy@mailinator.com",
        identifier: "87657257008",
    };
    // 4. Specify the element that holds the namespace of the issuer
    const xmlNamespaceModel = {
        prefix: "dip",
        uri: "http://portal.mec.gov.br/diplomadigital/arquivos-em-xsd",
    };
    // 5. The fields 'DadosDiploma' and 'DadosRegistro' and the entire XML file must be signed
    const xadesOptionsDegreeData = {
        signatureType: signer_node_client_1.XadesSignatureTypes.XmlElement,
        elementToSignIdentifierType: signer_node_client_1.XadesElementIdentifierTypes.XPath,
        elementToSignIdentifier: "//dip:DadosDiploma",
        insertionOption: signer_node_client_1.XadesInsertionOptions.AppendChild,
    };
    const xadesOptionsModelRegisterData = {
        signatureType: signer_node_client_1.XadesSignatureTypes.XmlElement,
        elementToSignIdentifierType: signer_node_client_1.XadesElementIdentifierTypes.XPath,
        elementToSignIdentifier: "//dip:DadosRegistro",
        insertionOption: signer_node_client_1.XadesInsertionOptions.AppendChild,
    };
    const xadesOptionsModelFull = {
        signatureType: signer_node_client_1.XadesSignatureTypes.FullXml
    };
    // 6. Each signature requires its own flow action
    const degreeDataAction = {
        type: signer_node_client_1.FlowActionType.Signer,
        user: participantUserOne,
        xadesOptions: xadesOptionsDegreeData
    };
    const registerDataAction = {
        type: signer_node_client_1.FlowActionType.Signer,
        user: participantUserTwo,
        xadesOptions: xadesOptionsModelRegisterData
    };
    const flowActionCreateModelFull = {
        type: signer_node_client_1.FlowActionType.Signer,
        user: participantUserThree,
        xadesOptions: xadesOptionsModelFull
    };
    // 7. Send the document create request
    const documentRequest = {
        newFolderName: "New folder",
        files: [fileUploadModel],
        flowActions: [degreeDataAction, registerDataAction, flowActionCreateModelFull],
        xmlNamespaces: [xmlNamespaceModel]
    };
    documentApi.apiDocumentsPost(documentRequest).then((res) => {
        console.log("Document id: ", res.data[0].documentId);
    });
});
